File: scripts/parse_elapsed.rb
#!/usr/bin/ruby
#
# Usage:
#
# $ ./scripts/runtests.py -g onnx_real --show_log |& tee log
# $ ./scripts/parse_elapsed.rb log

tests = {}

cur_test = nil
File.readlines(ARGV[0]).each do |line|
  if line =~ /^Running for out\/onnx_real_(.*?)\//
    cur_test = $1
  elsif line =~ /^Elapsed: (\d+\.\d+)/
    tests[cur_test] = $1.to_f
  end
end

tests.each do |name, elapsed|
  puts "#{name} #{elapsed}"
end

File: scripts/parse_elapsed.py
#!/usr/bin/python3
#
# Usage:
#
# $ ./scripts/runtests.py -g onnx_real --show_log |& tee log
# $ ./scripts/parse_elapsed.py log

import re
import sys

tests = {}

cur_test = None
with open(sys.argv[1]) as f:
    for line in f:
        m = re.match(r'^Running for out/onnx_real_(.*?)/', line)
        if m:
            cur_test = m.group(1)
            continue
        m = re.match(r'^Elapsed: (\d+\.\d+)', line)
        if m:
            tests[cur_test] = float(m.group(1))

for name, elapsed in sorted(tests.items()):
    print('%s %s' % (name, elapsed))
